docs/src/man/basics.md

```diff
@@ -0,0 +1,2002 @@
+```
+
+As mentioned in the title that we will discuss `setindex!` for *DataFrame*, *SubDataFrame*, *DataFrameRow*. So here
+we are talking about *SubDataFrame* and to create a *SubDataFrame* we need of `view` function. To know more about 
+`view` please have a look to **Views** section of this tutorial. 
+
+```jldoctest dataframe
+julia> sdf = view(df1, :, 2:3) # Column subsetting
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ male             2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ male             1
+
+julia> sdf[2, :Sex] = "female" # set value of col `:Sex` in second row to `female` in-place
+"female"
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ female           2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ male             1
+
+julia> sdf[6, 1:2] = ["female", 3] # set value of multiple columns
+2-element Vector{Any}:
+  "female"
+ 3
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ female           2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ female           3
+```
+
+In the above examples we have talked about *SubDataFrame*. Now in the given example we set the value 
+of 3rd-column which is `:Job` in our data frame `df1` with `[4, 5, 7, 8, 2, 1]`:
+
+```jldoctest dataframe
+julia> df1[:, 3] = [4, 5, 7, 8, 2, 1]
+6-element Vector{Int64}:
+ 4
+ 5
+ 7
+ 8
+ 2
+ 1
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    80  male             4
+   2 │    98  female           5
+   3 │    78  male             7
+   4 │    95  transgender      8
+   5 │    78  female           2
+   6 │    89  female           1
+```
+
+Column `:Age` is replaced freshly allocated vector because of broadcasting assignment:
+
+```jldoctest dataframe
+julia> df1[!, :Age] .= [85, 89, 78, 58, 96, 68] # col `:Age` is replaced freshly allocated vector
+6-element Vector{Int64}:
+ 85
+ 89
+ 78
+ 58
+ 96
+ 68
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    85  male             4
+   2 │    89  female           5
+   3 │    78  male             7
+   4 │    58  transgender      8
+   5 │    96  female           2
+   6 │    68  female           1
+```
+
+Note that if column name `:Customers` and `:City` is not present in `df1` then using `!` and `:` are equivalent. The 
+major difference between in-place and replace operations is that replacing columns is needed if new values have a 
+different type than the old ones. For instance here `!` works and `:` fails:
```

Comment:
in the example you give both operations work.

Reply:
Because in explanation I had explained if columns `:Customers` and `:City` are not present already then both will work. Am I correct here?

Reply:
You write:

>  For instance here `!` works and `:` fails:

I do not see where `:` fails in your example.

Also in line 1117 you now write that column is "allocated in-place", which is an oxymoron (i.e. if we allocate it cannot be in-place). Simply a new column is allocated.

Finally - as we discussed - please do not add comments in the code. Comments should be in markdown. Thank you!